qwalk_worker.py

```diff
@@ -197,8 +204,14 @@ def rc_get_ips(creds: Creds) -> Sequence[str]:
     def run(self) -> None:
         if not os.path.exists("old-queue.txt"):
             self.run_task.work_start(self)
-            rc = RestClient(self.creds["QHOST"], self.creds.get("QPORT", REST_PORT))
-            rc.login(self.creds["QUSER"], self.creds["QPASS"])
+            if self.creds["QTOKEN"]:
+                rc = RestClient(
+                    self.creds["QHOST"], self.creds.get("QPORT", REST_PORT), Credentials(self.creds["QTOKEN"])
+                    )
+            else:
+                rc = RestClient(self.creds["QHOST"], self.creds.get("QPORT", REST_PORT))
+                rc.login(self.creds["QUSER"], self.creds["QPASS"])
+            
```

Comment:
It looks like this could be a function too; accept the creds object and produce a RestClient